projects/ipdg_stokes/post_processing/norms.cc

```diff
@@ -0,0 +1,92 @@
+#include "norms.h"
+
+#include <iostream>
+#include <vector>
+
+#include <lf/base/base.h>
+#include <lf/geometry/geometry.h>
+#include <lf/mesh/utils/utils.h>
+#include <lf/quad/quad.h>
+#include <lf/quad/quad_rule.h>
+
+#include <utils.h>
+
+namespace projects::ipdg_stokes::post_processing {
+
+double L2norm(const std::shared_ptr<const lf::mesh::Mesh> &mesh,
+              const std::function<Eigen::Vector2d(const lf::mesh::Entity &,
+                                                  const Eigen::Vector2d &)> &f,
+              unsigned quadrule_order) {
+  const lf::quad::QuadRule quadrule =
+      lf::quad::make_QuadRule(lf::base::RefEl::kTria(), quadrule_order);
+  double norm2 = 0;
+  for (const auto &entity : mesh->Entities(0)) {
+    const auto geom = entity.Geometry();
+    const Eigen::MatrixXd points = geom->Global(quadrule.Points());
+    const Eigen::VectorXd weights =
+        (geom->IntegrationElement(quadrule.Points()).array() *
+         quadrule.Weights().array())
+            .matrix();
+    for (lf::base::size_type i = 0; i < quadrule.NumPoints(); ++i) {
+      norm2 += weights[i] * f(entity, points.col(i)).squaredNorm();
+    }
+  }
+  return std::sqrt(norm2);
```

Comment:
I think the body of this function is more or less equivalent to (assuming that f is a mesh function):
```
return std::sqrt(lf::uscalfe::IntegrateMeshFunction(mesh, f, lf::uscalfe::squaredNorm(quadrule_order)));
```
? ;)